pandas/core/frame.py

```diff
@@ -5401,6 +5401,94 @@ def _construct_result(self, result) -> "DataFrame":
         out.columns = self.columns
         return out
 
+    @Appender(
+        """
+Returns
+-------
+DataFrame
+    DataFrame that shows the differences stacked side by side.
+
+See Also
+--------
+Series.differences: Show differences.
+
+Examples
+--------
+>>> df = pd.DataFrame(
+...     {
+...         "col1": ["a", "a", "b", "b", "a"],
+...         "col2": [1.0, 2.0, 3.0, np.nan, 5.0],
+...         "col3": [1.0, 2.0, 3.0, 4.0, 5.0]
+...     },
+...     columns=["col1", "col2", "col3"],
+... )
+>>> df
+  col1  col2  col3
+0    a   1.0   1.0
+1    a   2.0   2.0
+2    b   3.0   3.0
+3    b   NaN   4.0
+4    a   5.0   5.0
+
+>>> df2 = df.copy()
+>>> df2.loc[0, 'col1'] = 'c'
+>>> df2.loc[2, 'col3'] = 4.0
+>>> df2
+  col1  col2  col3
+0    c   1.0   1.0
+1    a   2.0   2.0
+2    b   3.0   4.0
+3    b   NaN   4.0
+4    a   5.0   5.0
+
+Stack the differences on columns
+
+>>> df.differences(df2)
+  col1       col3
+  self other self other
```

Comment:
Do we like `self` and `other`? Will users be familiar enough with `self`? Or should we use `left` and `right`?

Reply:
I mentioned this in the first message of this PR. 

I used the (self, other) pair as I think they naturally match with the parameter names and the concept of comparing `self` with `other`. Although in some cases (left, right) or (old, new) may be slight more intuitive for users, the key issue is that when user choose to stack results on `axis=0`, in which case the results are stacked vertically and `left-right` pair looks extremely weird. The `old-new` pair is also not ideal as sometimes users really are not comparing old data with new data, but simply comparing 2 different sets of stats.